lib/commands/deploy.ts

```diff
@@ -8,44 +8,40 @@ import Future = require("fibers/future");
 import iOSDeploymentValidatorLib = require("./../validators/ios-deployment-validator");
 import commandParams = require("../common/command-params");
 
+export class DeployHelper implements IDeployHelper {
+	constructor(protected $devicesServices: Mobile.IDevicesServices,
+		protected $logger: ILogger,
+		protected $fs: IFileSystem,
+		protected $project: Project.IProject,
+		protected $buildService: Project.IBuildService,
+		protected $liveSyncService: ILiveSyncService,
+		protected $errors: IErrors) { }
 
-export class DeployCommand implements ICommand {
-	constructor(private $devicesServices: Mobile.IDevicesServices,
-		private $logger: ILogger,
-		private $fs: IFileSystem,
-		private $project: Project.IProject,
-		private $buildService: Project.IBuildService,
-		private $commandsService: ICommandsService,
-		private $errors: IErrors,
-		private $stringParameter: ICommandParameter) { }
-
-	public allowedParameters = [this.$stringParameter];
-
-	public execute(args: string[]): IFuture<void> {
+	public deploy(platform?: string): IFuture<void> {
 		return ((): void => {
 			this.$project.ensureProject();
 
 			if (!this.$project.capabilities.deploy) {
 				this.$errors.fail("You will be able to deploy %s based applications in a future release of the Telerik AppBuilder CLI.", this.$project.projectData.Framework);
 			}
-			if (args[0] && !MobileHelper.isPlatformSupported(args[0])) {
-				this.$errors.fail("On your current OS, you cannot deploy apps on connected %s devices.", MobileHelper.normalizePlatformName(args[0]));
+			if(platform && !MobileHelper.isPlatformSupported(platform)) {
+				this.$errors.fail("On your current OS, you cannot deploy apps on connected %s devices.", MobileHelper.normalizePlatformName(platform));
 			}
 
-			this.deployCore(args).wait();
+			this.deployCore(platform).wait();
 		}).future<void>()();
 	}
 
-	private deployCore(args: string[]): IFuture<void> {
+	private deployCore(platform: string): IFuture<void> {
 		return ((): void => {
 			if (options.companion) {
 				this.$logger.warn("No deployment necessary when using AppBuilder companion." +
 					" Use the `livesync` command instead to avoid this warning.");
-				this.$commandsService.executeCommandUnchecked("livesync", args).wait();
+				this.$liveSyncService.livesync(platform).wait();
```

Comment:
Our current code is working this way. The idea is to tell the customers that ther's a difference between deploy and livesync commands, but still do the correct work when they call deploy --companion.
I think we should keep it this way.